encoder/json.go

```diff
@@ -271,8 +266,8 @@ func fillCommonFormatKey(e *types.CommonFormatEvent, row *[]interface{}, s *type
 
 func fillCommonFormatFields(c *types.CommonFormatEvent, row *[]interface{}, schema *types.TableSchema, filter []int) {
 	f := make([]types.CommonFormatField, 0, len(schema.Columns))
-	for i, j := 0, 0; i < len(schema.Columns); i++ {
-		if filteredField(filter, i, &j) {
+	for i := 0; i < len(schema.Columns); i++ {
+		if filter[i] == -1 {
 			continue
 		}
```

Comment:
warning: cyclomatic complexity 16 of function fillCommonFormatFields() is high (> 15) (gocyclo)